pkg/registry/apis/datasource/sub_query.go

```diff
@@ -73,9 +73,30 @@ func (r *subQueryREST) Connect(ctx context.Context, name string, opts runtime.Ob
 
 		ctx = backend.WithGrafanaConfig(ctx, pluginCtx.GrafanaConfig)
 		ctx = contextualMiddlewares(ctx)
+
+		// forward expected headers, log unexpected ones (but still forward for now)
+		headers := make(map[string]string)
+		// headers are case insensitive, however some datasources still check for camel casing so we have to send them camel cased
+		expectedHeaders := map[string]string{
+			"fromalert":      "FromAlert",
+			"content-type":   "Content-Type",
+			"content-length": "Content-Length",
+			"user-agent":     "User-Agent",
+			"accept":         "Accept",
+		}
+		for k, v := range req.Header {
+			headerToSend, ok := expectedHeaders[strings.ToLower(k)]
+			if ok {
+				headers[headerToSend] = v[0]
+			} else {
+				headers[k] = v[0]
+				r.builder.log.Warn(fmt.Sprintf("datasource received an unexpected header: %s, forwarding it for now however this is likely to change in the future", k))
```

Comment:
should we be looking at an exclude list here to ensure that we aren't forwarding sensitive information like grafana auth related headers to an external datasource API request?

Reply:
hmm maybe? I'm actually not very familiar with what headers we do or don't want to allow. Since we don't send any currently, perhaps it's best to just stick with not sending any unless they are on an allow list. We have the log line so hopefully that can help us debug anything we are missing.